### PR TITLE
[SYCLomatic] Segmented sort keys

### PR DESCRIPTION
**Pull request by danhoeflinger**
Adding dpct::segmented_sort_keys functional mapping in the same style as dpct::segmented_sort_pairs.

Testing provided by https://github.com/oneapi-src/SYCLomatic-test/pull/232.

### CODE REVIEW

**Review by zhimingwang36 — APPROVED** [review]

**Comment by danhoeflinger** [comment]
I resolved the merge conflicts and force pushed.

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc [hunk 1]
```diff
@@ -1452,6 +1452,47 @@ inline void segmented_sort_pairs_by_parallel_sorts(
...
+      sycl::malloc_shared<offset_type>(nsegments, policy.queue());
+  auto host_accessible_offset_ends =
+      sycl::malloc_shared<offset_type>(nsegments, policy.queue());
+  // make offsets accessible on host
+  ::std::copy(::std::forward<_ExecutionPolicy>(policy), begin_offsets,
+              begin_offsets + nsegments, host_accessible_offset_starts);
+  ::std::copy(::std::forward<_ExecutionPolicy>(policy), end_offsets,
+              end_offsets + nsegments, host_accessible_offset_ends);
+
+  for (int i = 0; i < nsegments; i++) {
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: Use size_t or uint64_t here? 
- comparison between signed and unsigned integer expressions
- maybe int will overflow.

Comment (danhoeflinger) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: Good catch.  I will fix this.  I think the pair version has the same issue,  If there is no objection, I'll fix that in this same PR.

Comment (danhoeflinger) [reply clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc]: fixed.

**Review by timmiesmith — APPROVED** [review]
LGTM

**Pull request merged**